### PR TITLE
fix(AWS API Gateway) Enable detailed CloudWatch Metrics

### PR DESCRIPTION
**Pull request by s-kikuchi**
## Isuue 
#6112 

## Summary
I have changed code to be able to enable detailed cloud watch metrics for api gateway.
In my PR, if you define serverless.yml like below, api gateway enables detailed cloud watch metrics.

```
:
provider: aws
  metrics:
    restApi: true
:
```

However, I'm not sure this format in serverless.yml is really correct, so please let me know if I did something wrong.

### CODE REVIEW

**Comment by s-kikuchi** [comment]
@medikoo 
Thank you for your kind suggestions.
- I have changed serverless.yml format to be able to enable detailed cloud watch metrics like below.

```yml
provider:
  apiGateway:
    metrics: true
```

- I have added how to configure for detailed cloud watch metrics in md.

If you have any other requests or questions, please let me know.

File: lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js [hunk 1]
```diff
@@ -216,6 +223,17 @@ function handleTracing() {
...
 }
 
+function handleMetrics() {
+  if (!this.hasMetricsConfigured) return;
+  const metricsEnabled = this.state.service.provider.apiGateway.metrics;
+
+  let operation = { op: 'replace', path: '/*/*/metrics/enabled', value: 'false' };
+  if (metricsEnabled) {
+    operation = { op: 'replace', path: '/*/*/metrics/enabled', value: 'true' };
+  }
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js:233]: It'll be nicer to make it one liner as:

```javascript
operation = { op: 'replace', path: '/*/*/metrics/enabled', value: metricsEnabled ? 'true' : 'false' };
```

**Comment by sajithneyo** [comment]
Waiting for PR to be merged

File: lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js (hunk 1 above)
Comment (s-kikuchi) [reply lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js]: @medikoo 
Thank you for your suggestion. You're right.
I have just changed the code like below,

https://github.com/serverless/serverless/pull/7754/files#diff-b8e198657dea88d759b31561660e36c4R230-R234

The reason why the code I have changed is 5 lines even if your request code is 1 line is prettier automatically changed it.

Please check it out when you have time.

**Comment by roenschg** [comment]
@s-kikuchi Why is this part of the provider.apiGateway and not part of the provider.logs.restApi configuration?

All other similar options that can be found on the same menu in AWS like acessLogging or level are placed under provider.logs.restApi.

**Comment by medikoo** [comment]
> @s-kikuchi Why is this part of the provider.apiGateway and not part of the provider.logs.restApi configuration?

This is about metrics not logs. AWS keeps both in same panel, but it doesn't mean that AWS keeps metrics in logs panel :)

**Pull request merged**